jobs: rename scheduled job name to label

dt opened the pull request:
Name implies unique identification and has already led to confusion
in our initial testing and QA. Calling it label feels like it establishes
more accurate user expectations.

Release note: none.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/53129)
<!-- Reviewable:end -->

miretskiy reviewed (approved):
> Reviewed 15 of 15 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/53129)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @dt)

___
*[pkg/clusterversion/cockroach_versions.go, line 587 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/53129#-MFBzOomAt9YGAMYSZqz:-MFBzOomAt9YGAMYSZr-:b-fq1d48) ([raw file](https://github.com/cockroachdb/cockroach/blob/63216a76bca4e96a1dda7e7635f1c4d83932c358/pkg/clusterversion/cockroach_versions.go#L587)):*
> ```Go
> 	},
> 	{
> 		// VersionBox2DType enables the use of the box2d type.
> ```

comment wrong?


<!-- Sent from Reviewable.io -->

dt committed:
jobs: rename scheduled job name to label

Name implies unique identification and has already led to confusion
in our initial testing and QA. Calling it label feels like it establishes
more accurate user expectations.

Release note: none.
